tests/test-cmd/test.sign.js

```diff
@@ -0,0 +1,127 @@
+/* @flow */
+import path from 'path';
+import {describe, it} from 'mocha';
+import {assert} from 'chai';
+import sinon from 'sinon';
+
+import {withTempDir} from '../../src/util/temp-dir';
+import {basicManifest} from '../test-util/test.manifest';
+import completeSignCommand from '../../src/cmd/sign';
+import {makeSureItFails, fixturePath} from '../helpers';
+
+
+describe('sign', () => {
+
+  function getStubs() {
+    const signingConfig = {
+      apiKey: 'AMO JWT issuer',
+      apiSecret: 'AMO JWT secret',
+      apiUrlPrefix: 'http://not-the-real-amo.com/api/v3',
+    };
+
+    const buildResult = {
+      extensionPath: '/tmp/built-web-extension.xpi',
+    };
+    const build = sinon.spy(() => Promise.resolve(buildResult));
+
+    const signingResult = {
+      success: true,
+      downloadedFiles: [],
+    };
+    const signAddon = sinon.spy(() => Promise.resolve(signingResult));
+
+    return {
+      signingConfig,
+      build,
+      buildResult,
+      signAddon,
+      signingResult,
+      preValidatedManifest: basicManifest,
+    };
+  }
+
+  /*
+   * Run the sign command with stubs for all dependencies.
+   */
+  function sign(artifactsDir, stubs) {
+    return completeSignCommand({
+      sourceDir: '/fake/path/to/local/extension',
+      artifactsDir,
+      ...stubs.signingConfig,
+    }, {
+      ...stubs,
+    });
+  }
+
+  it('builds and signs an extension', () => withTempDir(
+    // This test only stubs out the signer in an effort to integrate
+    // all other parts of the process.
+    (tmpDir) => {
+      let stubs = getStubs();
+      const artifactsDir = path.join(tmpDir.path(), 'artifacts');
+      return completeSignCommand(
+        {
+          sourceDir: fixturePath('minimal-web-ext'),
+          artifactsDir,
+          ...stubs.signingConfig,
+        }, {
+          signAddon: stubs.signAddon,
+        })
+        .then((result) => {
+          assert.equal(result.success, true);
+          // Do a sanity check that a built extension was passed to the
+          // signer.
+          assert.include(stubs.signAddon.firstCall.args[0].xpiPath,
+                         'minimal_extension-1.0.xpi');
+        });
+    }
+  ));
+
+  it('returns a signing result', () => withTempDir(
+    (tmpDir) => {
+      let stubs = getStubs();
+      return sign(tmpDir.path(), stubs)
+        .then((realResult) => {
+          assert.deepEqual(realResult, stubs.signingResult);
+        });
+    }
+  ));
+
+  it('calls the add-on signer', () => withTempDir(
+    (tmpDir) => {
+      let stubs = getStubs();
+      return sign(tmpDir.path(), stubs)
+        .then(() => {
+          assert.equal(stubs.signAddon.called, true);
+          assert.equal(stubs.signAddon.firstCall.args[0].apiKey,
```

Comment:
Nit, how about assigning `stubs.signAddon.firstCall.args[0]` to a shorter variable name (e.g. `let signAddonArg = ...`) to make this sequence of assertions a bit more readable?